arcgis-ios-sdk-samples/Augmented reality/Navigate in AR/NavigateARRoutePlannerViewController.swift

```diff
@@ -0,0 +1,240 @@
+    
+    /// Create the start and destination stops for the navigation.
+    ///
+    /// - Returns: An array of `AGSStop` objects.
+    func makeStops() -> [AGSStop] {
+        let stop1 = AGSStop(point: self.startPoint!)
+        stop1.name = "Start"
+        let stop2 = AGSStop(point: self.endPoint!)
+        stop2.name = "Destination"
+        return [stop1, stop2]
+    }
+    
+    /// A wrapper function for operations after the route is solved by an `AGSRouteTask`.
+    ///
+    /// - Parameters:
+    ///   - routeResult: The result from `AGSRouteTask.solveRoute(with:completion:)`.
+    ///   - error: The error from `AGSRouteTask.solveRoute(with:completion:)`.
+    func didSolveRoute(with routeResult: AGSRouteResult?, error: Error?) {
+        if let error = error {
+            self.presentAlert(error: error)
+            self.setStatus(message: "Failed to solve route.")
+        } else if let result = routeResult, let firstRoute = result.routes.first {
+            self.routeResult = result
+            let routeGraphic = AGSGraphic(geometry: firstRoute.routeGeometry, symbol: nil)
+            self.routeGraphicsOverlay.graphics.add(routeGraphic)
+            self.setStatus(message: "Tap camera to start navigation.")
+        }
+    }
+    
+    // MARK: UI
+    
+    func setStatus(message: String) {
+        statusLabel.text = message
+    }
+    
+    // MARK: Actions
+    
+    @IBAction func reset() {
+        routeGraphicsOverlay.graphics.removeAllObjects()
+        stopGraphicsOverlay.graphics.removeAllObjects()
+        routeParameters.clearStops()
+        routeResult = nil
+        startPoint = nil
+        endPoint = nil
+        resetButtonItem.isEnabled = false
+        navigateButtonItem.isEnabled = false
+        setStatus(message: "Tap to place a start point.")
+    }
+    
+    // MARK: UIViewController
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        // Add the source code button item to the right of navigation bar.
+        (self.navigationItem.rightBarButtonItem as? SourceCodeBarButtonItem)?.filenames = [
+            "NavigateARRoutePlannerViewController",
+            "NavigateARNavigatorViewController",
+            "NavigateARCalibrationViewController"
+        ]
+        // Avoid overlapping status label and map content.
+        mapView.contentInset.top = 2 * statusLabel.font.lineHeight
+        
+        // Configure the authentication manager to show the OAuth dialog.
+        AGSAuthenticationManager.shared().delegate = self
+        AGSAuthenticationManager.shared().oAuthConfigurations.add(oAuthConfiguration)
+        
+        routeTask.load { [weak self] (error: Error?) in
+            guard let self = self else { return }
+            if let error = error {
+                self.presentAlert(error: error)
+                self.setStatus(message: "Failed to load route task. Check your connection or credentials.")
+                return
+            }
```

Comment:
Instead of an explicit return here, I recommend moving what follows into an `else` clause of the `if` statement.

Reply:
Haha at first I want to make the indentation look better. 😅